fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -340,6 +340,17 @@ private boolean isSurveyCategory(final Integer category) {
         return category.equals(DataTableApiConstant.CATEGORY_PPI);
     }
 
+    private JsonElement addColumn(final String name, final String dataType, final boolean isMandatory, final Integer length) {
```

Comment:
Do you think we could call this method "newColumn" or something instead since it's not really adding a new column anywhere but returns just part of a JsonObject?

I feel it's a little weird when I read the code below
`columns.add(addColumn("created_at", "DateTime", false, null));`